server/app/services/program/predicate/PredicateValue.java

```diff
@@ -28,6 +28,10 @@ public static PredicateValue of(long value) {
     return create(String.valueOf(value), OperatorRightHandType.LONG);
   }
 
+  public static PredicateValue of(double value) {
+    return create(String.valueOf(value), OperatorRightHandType.LONG);
```

Comment:
question (non-blocking):  Do we need `OperatorRightHandType.DOUBLE` here?

Reply:
Yeah, good catch - updated!